### PR TITLE
nerdfetch: init at 8.1.0

### PR DESCRIPTION
**Pull request by ByteSudoer**
## Description of changes
Nerdfetch: A POSIX *nix (Linux, macOS, Android, BSD, etc) fetch script using Nerdfonts (and others).
Homepage: https://codeberg.org/thatonecalculator/NerdFetch

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitea
+}:
+stdenv.mkDerivation rec {
+  pname = "nerdfetch";
+  version = "8.1.0";
```
Comment (jopejoe1) [inline pkgs/by-name/ne/nerdfetch/package.nix:7]: Newest version is 8.1.1 and seems to only be released on github https://github.com/ThatOneCalculator/NerdFetch/releases

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+  installPhase = ''
+    mkdir -p $out/bin
+      cp $src/nerdfetch $out/bin
+  '';
+
```
Comment (jopejoe1) [inline pkgs/by-name/ne/nerdfetch/package.nix:24]: Maybe add an update script

```suggestion

  passthru.updateScript = gitUpdater {
    rev-prefix = "v";
  };
  
```

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitea
+}:
+stdenv.mkDerivation rec {
```
Comment (jopejoe1) [inline pkgs/by-name/ne/nerdfetch/package.nix:5]: We are currently in the procces of switching from rec to using finalAttrs

```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  pname = "nerdfetch";
+  version = "8.1.0";
+
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "thatonecalculator";
+    repo = "NerdFetch";
+    rev = "v${version}";
```
Comment (jopejoe1) [inline pkgs/by-name/ne/nerdfetch/package.nix:13]: See above regarding switch to finalAttrs
```suggestion
    rev = "v${finalAttrs.version}";
```

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  meta = with lib;{
+    description = "A POSIX *nix (Linux, macOS, Android, *BSD, etc) fetch script using Nerdfonts";
+    homepage = "https://codeberg.org/thatonecalculator/nerdfetch";
+    maintainers = with maintainers; [ ByteSudoer ];
+    license = licenses.mit;
+    mainProgram = "nerdfetch";
+  };
+}
```
Comment (jopejoe1) [inline pkgs/by-name/ne/nerdfetch/package.nix:32]: See above regarding switch to finalAttrs
```suggestion
})
```

**Review by jopejoe1 — CHANGES_REQUESTED** [review]
Some small nitpicks but not blockers.

Also update your first commit mesaage to 
`maintainers: add ByteSudoer`

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      cp $src/nerdfetch $out/bin
+  '';
+
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "v";
+  };
+
+  meta = with lib;{
+    description = "A POSIX *nix (Linux, macOS, Android, *BSD, etc) fetch script using Nerdfonts";
+    homepage = "https://codeberg.org/thatonecalculator/nerdfetch";
```
Comment (Aleksanaa) [inline pkgs/by-name/ne/nerdfetch/package.nix:31]:
```suggestion
    homepage = "https://github.com/ThatOneCalculator/NerdFetch";
```

The codeberg repo is just a mirror which is not up to date.

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 7]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, gitUpdater
+}:
+stdenv.mkDerivation (finalAttrs: {
```
Comment (Aleksanaa) [inline pkgs/by-name/ne/nerdfetch/package.nix:6]:
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
```

Test if it works.

File: pkgs/by-name/ne/nerdfetch/package.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "v";
+  };
+
+  meta = with lib;{
+    description = "A POSIX *nix (Linux, macOS, Android, *BSD, etc) fetch script using Nerdfonts";
+    homepage = "https://codeberg.org/thatonecalculator/nerdfetch";
+    maintainers = with maintainers; [ ByteSudoer ];
+    license = licenses.mit;
+    mainProgram = "nerdfetch";
```
Comment (Aleksanaa) [inline pkgs/by-name/ne/nerdfetch/package.nix:34]:
```suggestion
    mainProgram = "nerdfetch";
    platforms = platforms.unix;
```

https://github.com/ThatOneCalculator/NerdFetch?tab=readme-ov-file#dependencies

File: pkgs/by-name/ne/nerdfetch/package.nix (hunk 7 above)
Comment (ByteSudoer) [reply pkgs/by-name/ne/nerdfetch/package.nix]: Yes I made the changes you requested and i can confirm that it works.

**Review by jopejoe1 — CHANGES_REQUESTED** [review]
Please Drop all commits except for the one where you add the package c1606b559a78e3a4b15ae837345e95bd134238e3

**Review by jopejoe1 — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]
```
=> ./result/bin/nerdfetch 

      ___     aleksana@Aleksana-Laptop 
     (.. \    󱄅  NixOS 24.05 (Uakari)
     (<> |      6.8.0
    //  \ \   󰍛  8954/27757 MiB (32%)
   ( |  | /|  󰏔  738 (nix-env)
  _/\ __)/_)  󰅶  1 day, 9 hours, 17 mins
  \/-____\/     ██████████████████
```

**Pull request merged**